R/schambach_table.R

```diff
@@ -0,0 +1,157 @@
+#' 
+#' @param mapvizieR_obj mapvizieR object
+#' @param measurementscale_is target subject
+#' @param grade target grade
+#' @param subgroup_cols what subgroups to explore, default is by school
+#' @param pretty_names nicely formatted names for the column cuts used above.
+#' @param start_fws starting season
+#' @param start_academic_year starting academic year
+#' @param end_fws ending season
+#' @param end_academic_year ending academic year
+#' @param complete_obsv if TRUE, limit only to students who have BOTH a start
+#' and end score. default is FALSE.
+#' 
+#' @export
+
+schambach_table_1d <- function(
+  mapvizieR_obj, 
+  measurementscale_is,
+  grade,
+  subgroup_cols = c('end_schoolname'),
+  pretty_names = c('School Name'),
+  start_fws,
+  start_academic_year,
+  end_fws,
+  end_academic_year,
+  complete_obsv = FALSE
+) {
+  
+#   if (missing(subgroup_cols) | missing(pretty_names)) {
+#     subgroup_cols <- c('end_schoolname')
+#     pretty_names <- c('School Name')
+#   } else {
+#     subgroup_cols <- c('end_schoolname',subgroup_cols)
+#     pretty_names <- c('School Name',pretty_names)
+#   }
+  
+  #data validation and unpack
+  assertthat::assert_that(length(subgroup_cols) == length(pretty_names))
+  
+  #unpack the mapvizieR object and limit to desired students
+  roster <- mapvizieR_obj[['roster']]
+  
+  #limit to desired grade level
+  growth_df <- mapvizieR_obj[['growth_df']]
+  growth_df <- growth_df %>% 
+    dplyr::filter(
+      end_grade == grade,
+      measurementscale == measurementscale_is
+    )
+  
+  #just desired terms
+  this_growth <- growth_df %>%
+    dplyr::filter(
+      start_map_year_academic == start_academic_year,
+      start_fallwinterspring == start_fws,
+      end_map_year_academic == end_academic_year,
+      end_fallwinterspring == end_fws
+    )
+  
+  grades_present <- unique(this_growth$start_grade)
+  
+  if(length(grades_present) > 1) {
+    warning(
+      sprintf(paste0("%i distinct grade levels present in your data! NWEA ",
+                     "school growth study tables assume cohorts composed of students ",
+                     "of the *same grade level*.  quealy_subgroups will use the mean starting grade ",
+                     "level to calculate growth scores, but you are advised to check your data and  ",
+                     "attempt to use a cohort composed of students from the same grade."), 
+              length(grades_present))
+    )
+  }
+  
+  roster <- dplyr::left_join(
+    x = roster,
+    y = this_growth[ ,c('studentid','end_schoolname')],
+    by = 'studentid'
+  )
+  
+  roster %>% 
+    ensurer::ensure_that(
+      all(subgroup_cols %in% names(roster)) ~ "subgroup_cols must match column names in your mapvizieR roster."
+    )
+  
+  if (complete_obsv == TRUE) {
+    this_growth <- this_growth %>%
+      dplyr::filter(
+        complete_obsv == TRUE  
+      )
+  }
+  
+  group_summary <- function(grouped_df) {
```

Comment:
@chrishaid I did this the same way on quealy plots -- are you OK with this convention?  if it's function-specific and never going to get re-used (like here) I don't think we do ourselves any favors by extracting it from the context of the local function.  this gets my :ok: of approval.